misc-scripts/xref_mapping/XrefParser/EntrezGeneParser.pm

```diff
@@ -50,74 +50,29 @@ sub run {
   
 
   my $eg_io = $self->get_filehandle($file);
-  if ( !defined $eg_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
-  }
+  croak "ERROR: Could not open $file\n" unless defined $eg_io;
 
-  my %seen;
+  my $input_file = Text::CSV->new({ sep_char           => "\t",
+				    empty_is_undef     => 1,
+				    allow_loose_quotes => 1 # turns out file has unescaped quotes
+				  }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
 
+  # process header
+  $input_file->column_names( @{ $input_file->getline( $eg_io ) } );
```

Comment:
$csv->header($eg_io) might suffice for this, depending on the first line of your file.